src/librustdoc/html/static/main.js

```diff
@@ -564,40 +564,36 @@ function hideThemeButtonState() {
         }
     }());
 
-    function addSidebarCrates(crates) {
-        // Draw a convenient sidebar of known crates if we have a listing
-        if (window.rootPath === "../" || window.rootPath === "./") {
-            var sidebar = document.getElementsByClassName("sidebar-elems")[0];
-            if (sidebar) {
-                var div = document.createElement("div");
-                div.className = "block crate";
-                div.innerHTML = "<h3>Crates</h3>";
-                var ul = document.createElement("ul");
-                div.appendChild(ul);
-
-                for (var i = 0; i < crates.length; ++i) {
-                    var klass = "crate";
-                    if (window.rootPath !== "./" && crates[i] === window.currentCrate) {
-                        klass += " current";
-                    }
-                    var link = document.createElement("a");
-                    link.href = window.rootPath + crates[i] + "/index.html";
-                    link.className = klass;
-                    link.textContent = crates[i];
-
-                    var li = document.createElement("li");
-                    li.appendChild(link);
-                    ul.appendChild(li);
-                }
-                sidebar.appendChild(div);
-            }
-        }
-    }
-
     // delayed sidebar rendering.
     window.initSidebarItems = function(items) {
         var sidebar = document.getElementsByClassName("sidebar-elems")[0];
         var current = window.sidebarCurrent;
+        var isModule = getNakedUrl().endsWith("/index.html");
```

Comment:
This check might fail at the root module if the webserver redirects `/` to `/index.html`. In fact, this happens on doc.rust-lang.org causing https://doc.rust-lang.org/std/ to be the same as https://doc.rust-lang.org/std/index.html. 

This would cause `isModule` to be false, and lead to showing all items at the root level. You can reproduce it locally by running `python -m http.server`, which does this silent redirection.

Reply:
This is a good point, thanks!